crates/hyperqueue/src/dashboard/ui/screens/worker/worker_tasks_table.rs

```diff
@@ -0,0 +1,110 @@
+use crate::common::format::human_duration;
+use crate::dashboard::data::tasks_timeline::TaskInfo;
+use crate::dashboard::ui::terminal::DashboardFrame;
+use crate::dashboard::ui::widgets::table::{StatefulTable, TableColumnHeaders};
+use chrono::{DateTime, Local};
+use std::time::SystemTime;
+use tako::TaskId;
+use tui::layout::{Constraint, Rect};
+use tui::widgets::{Cell, Row};
+
+#[derive(Default)]
+pub struct WorkerTasksTable {
+    table: StatefulTable<WorkerTaskRow>,
+}
+
+impl WorkerTasksTable {
+    pub fn update(&mut self, tasks_info: Vec<&TaskInfo>) {
+        let mut rows = create_rows(tasks_info);
+        rows.sort_by_key(|task_row| task_row.task_state.len());
+        self.table.set_items(rows);
+    }
+
+    pub fn select_next_task(&mut self) {
+        self.table.select_next_wrap();
+    }
+
+    pub fn clear_selection(&mut self) {
+        self.table.clear_selection();
+    }
+
+    pub fn select_previous_task(&mut self) {
+        self.table.select_previous_wrap();
+    }
+
+    pub fn get_selected_item(&self) -> Option<TaskId> {
+        let selection = self.table.current_selection();
+        selection.map(|row| row.task_id)
+    }
+
+    pub fn draw(&mut self, rect: Rect, frame: &mut DashboardFrame) {
+        self.table.draw(
+            rect,
+            frame,
+            TableColumnHeaders {
+                title: "Tasks Info".to_string(),
```

Comment:
I suppose that `title` and `inline_help` could be converted to `&'static str` or at least `Cow<str>`.